### PR TITLE
OPENEUROPA-2096: ECL 2.0: Table

### PR DESCRIPTION
**Pull request by voidtek**
### Description

ECL 2.0: Table

### CODE REVIEW

File: templates/layout/page.html.twig [hunk 1]
```diff
@@ -33,7 +33,7 @@
 {% endblock %}
 
 {% block content %}
-<main class="ecl-u-pv-xl" id="main-content">
```
Comment (imanoleguskiza) [inline templates/layout/page.html.twig]: What is the reasoning behind this?

File: templates/status/status-messages.html.twig [hunk 2]
```diff
@@ -25,6 +25,7 @@
     'description': _self.message_list(messages),
     'variant': (type == 'status' ? 'success' : type),
     'extra_attributes': extra_attributes,
+    'extra_classes': 'ecl-u-mv-l',
```
Comment (imanoleguskiza) [inline templates/status/status-messages.html.twig]: I assume this is to ofset the class we removed on page.html, but why was that done?

File: oe_theme.theme [hunk 3]
```diff
@@ -1045,3 +1045,41 @@ function oe_theme_preprocess_language_selection_page_content(&$variables) {
...
+ */
+function oe_theme_preprocess_block__oe_theme_main_page_content(&$variables) {
+  $variables['attributes']['class'][] = 'ecl-u-mv-l';
+}
+
+/**
+ * Implements hook_preprocess_table().
+ */
+function oe_theme_preprocess_table(array &$variables): void {
+  $variables['zebra'] = 'true';
```
Comment (imanoleguskiza) [inline oe_theme.theme]: How come we are always setting the zebra to true?

File: templates/ui_patterns_library/patterns-overview-page.html.twig [hunk 4]
```diff
@@ -24,35 +24,32 @@
   } %}
 
   {% for pattern_name, pattern in patterns %}
+    <div class="ecl-u-mv-l ecl-u-border-color-grey-25 ecl-u-border-bottom">
```
Comment (imanoleguskiza) [inline templates/ui_patterns_library/patterns-overview-page.html.twig]: All of the changes done in this file seem unrelated to the porting the table component.

File: oe_theme.theme [hunk 5]
```diff
@@ -1045,3 +1045,41 @@ function oe_theme_preprocess_language_selection_page_content(&$variables) {
...
+
+  $name_col = [];
+  foreach ($variables['header'] as $key => &$cell) {
```
Comment (imanoleguskiza) [inline oe_theme.theme]: Defining the $key variable is not required.

File: oe_theme.theme [hunk 6]
```diff
@@ -1045,3 +1045,41 @@ function oe_theme_preprocess_language_selection_page_content(&$variables) {
...
+    $cell['label'] = $cell['content'];
+
+    if ($cell['attributes']['colspan']) {
+      $cell['colspan'] = (string) $cell['attributes']['colspan'];
+    }
+
+    for ($i = 1; $i <= (array_key_exists('colspan', $cell) ? intval($cell['colspan']) : 1); $i++) {
+      $cell_label = (array) $cell['label'];
+      $name_col[] = array_key_exists('#markup', $cell_label) ? $cell_label['#markup'] : $cell['label'];
```
Comment (imanoleguskiza) [inline oe_theme.theme]: Can we add a small comment as to why we are doing this? Something like: "The contents of a drupal cell can be a render array so we need to make sure we have an string to set as the 'data-ecl-table-header' property later"

File: templates/layout/page.html.twig (hunk 1 above)
Comment (voidtek) [reply templates/layout/page.html.twig]: done

File: templates/status/status-messages.html.twig (hunk 2 above)
Comment (voidtek) [reply templates/status/status-messages.html.twig]: removed

File: templates/ui_patterns_library/patterns-overview-page.html.twig (hunk 4 above)
Comment (voidtek) [reply templates/ui_patterns_library/patterns-overview-page.html.twig]: Removed.

File: oe_theme.theme (hunk 3 above)
Comment (voidtek) [reply oe_theme.theme]: Removed.

File: oe_theme.theme (hunk 5 above)
Comment (voidtek) [reply oe_theme.theme]: done.

File: oe_theme.theme (hunk 6 above)
Comment (voidtek) [reply oe_theme.theme]: done.

File: oe_theme.theme [hunk 7]
```diff
@@ -1044,3 +1044,35 @@ function oe_theme_preprocess_language_selection_page_content(&$variables) {
...
+    if ($cell['attributes']['colspan']) {
+      $cell['colspan'] = (string) $cell['attributes']['colspan'];
+    }
+
+    for ($i = 1; $i <= (array_key_exists('colspan', $cell) ? intval($cell['colspan']) : 1); $i++) {
+      $cell_label = (array) $cell['label'];
+      // The contents of a drupal cell can be a render array so we need to make
+      // sure we have an string to set as the 'data-ecl-table-header' property
+      // later.
+      $name_col[] = array_key_exists('#markup', $cell_label) ? $cell_label['#markup'] : $cell['label'];
```
Comment (imanoleguskiza) [inline oe_theme.theme]: I am still not sure about this approach. We can't really be sure that the cell content is going to be a simple markup render array. For all we know, this may be anything so I think we need to find a more open solution. I added a possible approach here: https://github.com/openeuropa/oe_theme/pull/332/files in which I render the array and extract all tags to end up with the actual text. Let me know what you think
